### DISCUSSION TITLE
Noding Basic SegmentStrings vs Noded SegmentStrings

### DISCUSSION THREAD

**Discussion by micycle1** [locationtech/jts#813] [Q&A] [answered]
Is there any difference/advantage in passing a list of `NodedSegmentString` vs `BasicSegmentString` to a `Noder` (`noder.computeNodes()`)?

**Comment by dr-jts** [comment, marked as answer]
Yes, there is.  I have just noticed that the Javadoc on `BasicSegmentString` is not correct, so thanks for asking this question (I will fix it pronto).

The difference is:
* `BasicSegmentString` allows accessing segments, but does not support adding nodes
* `NodedSegmentString` allows accessing segments, and also allows adding nodes to the segments, and then splitting the segstring at the nodes to produce a set of sub-strings. 

`Noder`s are used for both **detecting intersections** and for **computing noded arrangements**.  For **detection** it is fine to use `BasicSegmentString`.  This avoids the (small) overhead of creating a `SegmentNodeList` for each string.  But if full **noding** is required then `NodedSegmentString` must be used.

(Really `NodedSegmentString` could be a subclass of `BasicSegmentString` - not sure why this is not so.   And perhaps the `SegmentNodeList` should be created lazily, which would reduce the overhead to the point where only `NodedSegmentString` would be needed).

**Comment by dr-jts** [comment]
The Javadoc for `BasicSegmentString` and `NodedSegmentString` has been improved.
